### PR TITLE
Added Lambda function details to create-function response

### LINKED ISSUE #305: AWS Lambda - Create lambda function does not return function details
When creating a new lambda function using the CLI, the function is created, but the lambda details are not returned.

## Context

According to the [AWS Lambda API documentation](https://docs.aws.amazon.com/lambda/latest/dg/API_CreateFunction.html), creating a new lambda function will return the details of the function created.

## Process

1. Package the lambda into a zip archive.
2. Create a new lambda using AWS CLI command
`aws --endpoint-url=http://localhost:4574 --region=us-east-1 lambda create-function --function-name=myFunction --runtime=nodejs6.10 --role=arn:aws:iam:local --handler=lambda.handler --zip-file=fileb://lambda.zip`
3. The lambda function is created and the result printed on the screen.

## Expected result

```javascript
HTTP/1.1 201
Content-type: application/json

{
   "CodeSha256": "string",
   "CodeSize": number,
   "DeadLetterConfig": { 
      "TargetArn": "string"
   },
   "Description": "string",
   "Environment": { 
      "Error": { 
         "ErrorCode": "string",
         "Message": "string"
      },
      "Variables": { 
         "string" : "string" 
      }
   },
   "FunctionArn": "string",
   "FunctionName": "string",
   "Handler": "string",
   "KMSKeyArn": "string",
   "LastModified": "string",
   "MasterArn": "string",
   "MemorySize": number,
   "Role": "string",
   "Runtime": "string",
   "Timeout": number,
   "TracingConfig": { 
      "Mode": "string"
   },
   "Version": "string",
   "VpcConfig": { 
      "SecurityGroupIds": [ "string" ],
      "SubnetIds": [ "string" ],
      "VpcId": "string"
   }
}
```

## Current result

`{}`

### PR DESCRIPTION
**Pull request by migurski**
Added known function details to `create-function` response.

Sample call:

    aws --endpoint-url http://localhost:4574 lambda create-function --runtime python3.6 --role Nobody --handler lambda.score_plan --function-name 'WOMP' --zip-file fileb://lambda.zip --description 'the-desc' --timeout 30 --memory-size 128 --publish --dead-letter-config TargetArn=the-dlc

Sample output:

```
{
    "FunctionName": "WOMP",
    "FunctionArn": "arn:aws:lambda:us-east-1:000000000000:function:WOMP",
    "Runtime": "python3.6",
    "Role": "Nobody",
    "Handler": "lambda.score_plan",
    "Description": "the-desc",
    "Timeout": 30,
    "MemorySize": 128,
    "VpcConfig": {
        "SubnetIds": [
            null
        ],
        "SecurityGroupIds": [
            null
        ]
    },
    "DeadLetterConfig": {
        "TargetArn": "the-dlc"
    },
    "Environment": {
        "Variables": {},
        "Error": {}
    },
    "TracingConfig": {}
}
```

- [ ] Add tests if this makes sense.

Closes #305.

### CODE REVIEW

**Comment by whummer** [comment]
Thanks for this PR @migurski . Looks like there are a few lint errors, but otherwise looks good to me 👍

**Comment by migurski** [comment]
Fixed lint errors, trying tests again.

File: localstack/services/awslambda/lambda_api.py [hunk 1]
```diff
@@ -515,7 +517,21 @@ def create_function():
         lambda_arn_to_runtime[arn] = data['Runtime']
         lambda_arn_to_envvars[arn] = data.get('Environment', {}).get('Variables', {})
         result = set_function_code(data['Code'], lambda_name)
-        return result or jsonify({})
+        result.update({
+            "DeadLetterConfig": data['DeadLetterConfig'],
```
Comment (whummer) [inline localstack/services/awslambda/lambda_api.py]: This should be `data.get('DeadLetterConfig')`, as `DeadLetterConfig` is optional.

File: localstack/services/awslambda/lambda_api.py [hunk 2]
```diff
@@ -515,7 +517,21 @@ def create_function():
...
+            "Description": data['Description'],
```
Comment (whummer) [inline localstack/services/awslambda/lambda_api.py]: Should be `data.get('Description')`. Same for the other optional parameters below: `MemorySize` and `Timeout`

**Comment by whummer** [comment]
Awesome, thanks again for this PR! 👍

**Pull request merged**